src/main/java/org/opensearch/neuralsearch/search/query/HybridCollectorManager.java

```diff
@@ -127,14 +170,32 @@ public ReduceableSearchResult reduce(Collection<Collector> collectors) {
                 for (final Collector sub : (((MultiCollectorWrapper) collector).getCollectors())) {
                     if (sub instanceof HybridTopScoreDocCollector) {
                         hybridTopScoreDocCollectors.add((HybridTopScoreDocCollector) sub);
+                    } else if (sub instanceof HybridTopFieldDocSortCollector.SimpleFieldCollector) {
+                        hybridSortedTopDocCollectors.add((HybridTopFieldDocSortCollector.SimpleFieldCollector) sub);
+                    } else if (sub instanceof HybridTopFieldDocSortCollector.PagingFieldCollector) {
+                        hybridSortedTopDocCollectors.add((HybridTopFieldDocSortCollector.PagingFieldCollector) sub);
                     }
                 }
             } else if (collector instanceof HybridTopScoreDocCollector) {
                 hybridTopScoreDocCollectors.add((HybridTopScoreDocCollector) collector);
+            } else if (collector instanceof HybridTopFieldDocSortCollector.SimpleFieldCollector) {
+                hybridSortedTopDocCollectors.add((HybridTopFieldDocSortCollector.SimpleFieldCollector) collector);
+            } else if (collector instanceof HybridTopFieldDocSortCollector.PagingFieldCollector) {
+                hybridSortedTopDocCollectors.add((HybridTopFieldDocSortCollector.PagingFieldCollector) collector);
             } else if (collector instanceof FilteredCollector
                 && ((FilteredCollector) collector).getCollector() instanceof HybridTopScoreDocCollector) {
                     hybridTopScoreDocCollectors.add((HybridTopScoreDocCollector) ((FilteredCollector) collector).getCollector());
-                }
+                } else if (collector instanceof FilteredCollector
+                    && ((FilteredCollector) collector).getCollector() instanceof HybridTopFieldDocSortCollector.SimpleFieldCollector) {
+                        hybridSortedTopDocCollectors.add(
+                            (HybridTopFieldDocSortCollector.SimpleFieldCollector) ((FilteredCollector) collector).getCollector()
+                        );
+                    } else if (collector instanceof FilteredCollector
+                        && ((FilteredCollector) collector).getCollector() instanceof HybridTopFieldDocSortCollector.PagingFieldCollector) {
+                            hybridSortedTopDocCollectors.add(
+                                (HybridTopFieldDocSortCollector.PagingFieldCollector) ((FilteredCollector) collector).getCollector()
+                            );
+                        }
```

Comment:
should we validate that both regular score and sorted score collectors are not empty?